crates/holochain/src/core/workflow/sys_validation_workflow/validation_deps.rs

```diff
@@ -202,10 +315,63 @@ impl ValidationDependencyState {
         }
     }
 
+    pub(super) fn set_pending_validation_warranted(
+        &mut self,
+        action: SignedActionHashed,
+        op_type: ChainOpType,
+    ) {
+        if self.dependency_type == ValidationDependencyType::Action {
+            tracing::warn!("Attempted to set an action on a dependency that is not of type Action, this is a bug");
+            return;
+        }
+
+        match self.dependency {
+            None => {
+                self.dependency = Some(ValidationDependency {
+                    value: ValidationDependencyValue::Warranted(WarrantedDep::Pending(
+                        action, op_type,
+                    )),
+                    fetched_from: CascadeSource::Network,
+                });
+            }
+            _ => {
+                tracing::warn!("Attempted to set a record on a dependency that already has a value, this is a bug")
+            }
+        }
+    }
+
+    pub(super) fn set_validation_status_for_warranted(
+        &mut self,
+        validation_status: ValidationStatus,
+    ) {
+        if self.dependency_type == ValidationDependencyType::Action {
```

Comment:
Same here, checking for `!= ValidationDependencyValue::Warranted` rather?

Reply:
https://github.com/holochain/holochain/pull/5249/commits/9801bda0488c75727acee3e09527cb747eeaf3d9